### PR TITLE
RSDK-7881: Refactor Shutdown to use actual robot client

### PR DESCRIPTION
**Pull request by danielbotros**
### Description
RSDK-7881: Refactor Shutdown to use actual robot client
* This PR refactors the `Shutdown()` integration test to use the actual robot client so that we don't have to maintain two different sets of expected errors between the test and the actual client.

### CODE REVIEW

**Comment by cheukt** [comment]
removing myself since there's enough reviewers

File: web/server/entrypoint_test.go [hunk 1]
```diff
@@ -158,15 +159,11 @@ func TestShutdown(t *testing.T) {
...
-		defer func() {
-			test.That(t, conn.Close(), test.ShouldBeNil)
-		}()
-		rc := robotpb.NewRobotServiceClient(conn)
+		addr := "localhost:" + strconv.Itoa(port)
+		rc := robottestutils.NewRobotClient(t, testLogger, addr, time.Second)
 
 		testLogger.Info("Issuing shutdown.")
-		_, err = rc.Shutdown(context.Background(), &robotpb.ShutdownRequest{})
+		err = rc.Shutdown(context.Background())
```
Comment (maximpertsov) [inline web/server/entrypoint_test.go:166]:
```suggestion
		err = rc.Shutdown(context.Background())
		test.That(t, err, test.ShouldBeNil)
```

Comment (danielbotros) [reply web/server/entrypoint_test.go:166]: not guaranteed to be nil; err is checked with this line:
`test.That(t, isExpectedShutdownError(err, testLogger), test.ShouldBeTrue)`

Comment (maximpertsov) [reply web/server/entrypoint_test.go:166]: ah sorry, forgot about that

**Review by maximpertsov — APPROVED** [review]

**Review by benjirewis — APPROVED** [review]
Nice! Thanks for modifying the helper.

**Pull request merged**